### PR TITLE
refactor : removed globalc::hm and globalc::ufft

### LINKED ISSUE #1420: Misusing of GlobalC::hm in working code.
### Describe Current Status and Possible Solution

"GlobalC::hm" is a global class which is obsolete now.
When I try to delete this global class, I found that several code still used this class.

1. https://github.com/deepmodeling/abacus-develop/blob/develop/source/module_esolver/esolver_ks_pw.cpp#L88
```
GlobalC::hm.hpw.allocate(GlobalC::wf.npwx, GlobalV::NPOL, GlobalC::ppcell.nkb, GlobalC::rhopw->nrxx);
```
@Qianruipku can you solve this code coupling?

2. https://github.com/deepmodeling/abacus-develop/blob/develop/source/src_pw/sto_iter.cpp#L401
```
if(GlobalC::kv.nks > 1) GlobalC::hm.hpw.init_k(ik);
```
@Qianruipku I think HamiltPW::updateHk(ik) can replace it. 

3. https://github.com/deepmodeling/abacus-develop/blob/develop/source/module_hsolver/diago_lapack.cpp#L43
```
GlobalC::hm.diagH_LAPACK(GlobalV::NLOCAL, GlobalV::NBANDS, Htmp, Stmp, GlobalV::NLOCAL, en, hvec);
```
This is a public function just place in old class, function diagH_LAPACK() should move to diago_lapack.h file.

4. https://github.com/deepmodeling/abacus-develop/blob/develop/source/src_pw/wavefunc.cpp
Several lines used GlobalC::hm, which can be deleted now. @dyzheng 

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by wenfei-li**

### CODE REVIEW

File: source/module_hsolver/hsolver_lcao.cpp [hunk 1]
```diff
@@ -54,6 +53,10 @@ void HSolverLCAO::solveTemplate(hamilt::Hamilt* pHamilt,
     }
     else if (this->method == "lapack")
     {
+        //We are not supporting diagonalization with lapack
+        //until the obsolete globalc::hm is removed from
+        //diago_lapack.cpp
+        /*
```
Comment (dyzheng) [inline source/module_hsolver/hsolver_lcao.cpp:59]: This is related to compiling without MPI, can we ban this feature until Issue#1420 be solved?

File: source/src_pw/sto_iter.cpp [hunk 2]
```diff
@@ -398,7 +398,28 @@ void Stochastic_Iter::calHsqrtchi(Stochastic_WF& stowf)
     for(int ik = 0; ik < GlobalC::kv.nks; ++ik)
     {
         //init k
-        if(GlobalC::kv.nks > 1) GlobalC::hm.hpw.init_k(ik);
+        if(GlobalC::kv.nks > 1)
```
Comment (dyzheng) [inline source/src_pw/sto_iter.cpp:401]: Please review this change @Qianruipku

File: source/module_hsolver/hsolver_lcao.cpp (hunk 1 above)
Comment (wenfei-li) [reply source/module_hsolver/hsolver_lcao.cpp:59]: i think the lapack solver is already broken, unless someone has fixed it at some point ..

File: source/src_pw/sto_iter.cpp (hunk 2 above)
Comment (Qianruipku) [reply source/src_pw/sto_iter.cpp:401]: It is OK.

**Review by Qianruipku — APPROVED** [review]

**Review by dyzheng — APPROVED** [review]

**Pull request merged**